pkg/controller/authentication/job.go

```diff
@@ -91,6 +92,13 @@ func generateJobObject(instance *operatorv1alpha1.Authentication, scheme *runtim
 	reqLogger := log.WithValues("Instance.Namespace", instance.Namespace, "Instance.Name", instance.Name)
 	image := shatag.GetImageRef("IM_INITCONTAINER_IMAGE")
 	resources := instance.Spec.ClientRegistration.Resources
+
+	metaLabels := pkgCommon.MergeMap(map[string]string{"app": jobName}, instance.Spec.Labels)
+	podMetaLabels := map[string]string{
+		"app":                        jobName,
+		"app.kubernetes.io/instance": "oidc-client-registration",
+	}
+	podLabels := pkgCommon.MergeMap(podMetaLabels, instance.Spec.Labels)
```

Comment:
If the intent is for the `podMetaLabels` to be preserved, then the arguments should be reordered.